### DISCUSSION TITLE
Questions about using snap and scale together

### DISCUSSION THREAD

**Discussion by minetti0327** [konvajs/konva#1173] [Q&A]
Hello.
I'm using konva in my Company product!

I have a question.
Is there a way to maintain the starting point of a transform when scale and snap are used together?

sample( We actually use react-konva in my product. )
https://codesandbox.io/s/boring-brattain-zyu1i?file=/index.html

When transforming rect with sandbox, the starting point of the transform becomes the snapped point.
If you know how to fix it, please let me know!

**Comment by lavrton** [comment]
Can you clarify what behavior do you want? Snapping for resize may be it bit more complex.

**Comment by minetti0327** [comment]
Thanks for the reply.

There are cases where the transform stays at the starting point and cases where it doesn't.

case 1. transform stays at the starting point
![caputure-1](https://user-images.githubusercontent.com/28689245/123614224-1910c200-d83f-11eb-94bb-60511c3827ab.gif)

case 2. If the start point is a snap destination
![capture-2](https://user-images.githubusercontent.com/28689245/123614409-452c4300-d83f-11eb-9385-f85bb0ca2eac.gif)

Is it possible to keep the case 1?

**Comment by lavrton** [comment]
I still don't understand what you want. What do you mean by "When transforming rect with sandbox, the starting point of the transform becomes the snapped point."?

What kind of snap do you want when you resize if transformer? Snap what? Snap to what?

**Comment by minetti0327** [comment]
I apologize for not being able to give a clear explanation.

![capture1](https://user-images.githubusercontent.com/28689245/124054380-d7f5f900-da5c-11eb-878f-ba9fa44b978b.gif)

As shown in the gif image, when I scale, the image may be scaled starting from the snapped coordinates.
What I want to do is to keep the top-left scale.

It means that you don't want to center scale at the snap position of the middle of the rect.

you know how to make the transformer expand while maintaining the starting point (top-left) when it is first dragged, please let me know.
